parton/pdf.py

```diff
@@ -182,25 +195,23 @@ def __init__(self, name, member=0, pdfdir=None):
         meta, grids = self.pdfmember.load()
         self.pdfgrids = [PDFGrid.from_block(grid) for grid in grids]
 
-    def xfxQ(self, flavor, x, Q):
+    def xfxQ(self, flavor, x, Q, grid=True):
         """Return x*f(x) by specifying flavor, `x`, and factorization scale
         `Q` in GeV."""
-        return self.xfxQ2(flavor, x, Q**2)
+        return self.xfxQ2(flavor, x, Q**2, grid=grid)
 
-    def xfxQ2(self, flavor, x, Q2):
+    def xfxQ2(self, flavor, x, Q2, grid=True):
         """Return x*f(x) by specifying flavor, `x`, and factorization scale
         squared `Q2` in GeV^2."""
-        res = np.empty((np.size(x), np.size(Q2)))
-        res[:] = np.nan
+        if grid == True:
+            res = np.full((np.size(x), np.size(Q2)), np.nan)
+        else:
+            res = np.full(np.size(x), np.nan)
```

Comment:
This implicitly assumes that `flavor`, `x`, and `Q2` all have the same shape (and also that they are 1D), which should not be a requirement if broadcasting is used.